esphome/components/airthings_wave_plus/airthings_wave_plus.h

```diff
@@ -0,0 +1,90 @@
+#pragma once
+
+#include "esphome/core/component.h"
+#include "esphome/core/helpers.h"
+#include "esphome/core/preferences.h"
+#include "esphome/core/log.h"
+#include "esphome/core/application.h"
+#include "esphome/core/version.h"
+
+#include <BLEDevice.h>
+#include <algorithm>
+#include <iterator>
+
+namespace esphome {
+namespace airthings_wave_plus {
+
+static const char *TAG = "airthings_wave_plus";
+
+class AirthingsWavePlus : public Component {
+ public:
+  AirthingsWavePlus();
+
+  void dump_config() override;
+  void set_address(std::string address) { address_ = address; }
+  void set_update_interval(uint32_t update_interval);
+
+  void set_temperature(sensor::Sensor *temperature) { temperature_sensor_ = temperature; }
+  void set_radon(sensor::Sensor *radon) { radon_sensor_ = radon; }
+  void set_radon_long_term(sensor::Sensor *radon_long_term) { radon_long_term_sensor_ = radon_long_term; }
+  void set_humidity(sensor::Sensor *humidity) { humidity_sensor_ = humidity; }
+  void set_pressure(sensor::Sensor *pressure) { pressure_sensor_ = pressure; }
+  void set_co2(sensor::Sensor *co2) { co2_sensor_ = co2; }
+  void set_tvoc(sensor::Sensor *tvoc) { tvoc_sensor_ = tvoc; }
+
+ protected:
+  uint32_t connection_timeout_in_seconds_ = 30;
+  uint32_t update_interval_in_seconds_ = 300;
+
+  uint32_t update_count_ = 0;
+  bool connected_ = false;
+  bool connecting_ = false;
+  int32_t last_value_time_;
+  int32_t last_connect_time_ = connection_timeout_in_seconds_ * -1000;
```

Comment:
```suggestion
  uint32_t last_value_time_;
  uint32_t last_connect_time_;
```

These have to be unsigned, otherwise you'll get issues after 20 days of uptime (when the millis function rolls over)

`last_connect_time_` will have to be set in `setup()` to some suitable value (like the time at setup)

Reply:
Interesting indeed. ~40 days may not be particularly interesting either, all things considered. I could not find a 64 bits version of the counter, is there one to be used ?